### PR TITLE
Feature/sckan 300 b

### PR DESCRIPTION
**Pull request by afonsobspinto**
closes https://metacell.atlassian.net/browse/SCKAN-300

- Adds signals on origin, vias and destination changes that remove the graph state if it exists
- Updates statement viewset to return updated entity 
- Updates frontend to setStatement on origin changes and refresh statement on vias/destination changes:

https://github.com/user-attachments/assets/af235969-2ae9-45c2-95d3-48a4d3b1f6b1

### CODE REVIEW

File: frontend/src/components/Forms/StatementForm.tsx [hunk 1]
```diff
@@ -446,13 +450,13 @@ const StatementForm = (props: any) => {
           isPathBuilderComponent: true,
           InputIcon: DestinationIcon,
           onUpdate: async (selectedOption: string, formId: string) => {
-            const viaIndex = getConnectionId(formId, statement?.destinations);
+            const destinationIndex = getConnectionId(formId, statement?.destinations);
             const typeOption = selectedOption as TypeC11Enum;
-            if (viaIndex) {
```
Comment (afonsobspinto) [inline frontend/src/components/Forms/StatementForm.tsx:451]: I find it strange that this is not wrapped in a checkOwnership function. I suspect that the UI won't behave correctly when trying to change the type of a destination from a connectivity statement that I don't own. I didn't test this tho. cc @ddelpiano @Salam-Dalloul

File: frontend/src/components/Forms/StatementForm.tsx [hunk 2]
```diff
@@ -446,13 +450,13 @@ const StatementForm = (props: any) => {
...
-              api
-                .composerDestinationPartialUpdate(viaIndex, {
-                  type: typeOption,
-                })
+            if (destinationIndex) {
+              await api.composerDestinationPartialUpdate(destinationIndex, {
```
Comment (afonsobspinto) [inline frontend/src/components/Forms/StatementForm.tsx:456]: I find it strange that this is not wrapped in a checkOwnership function. I suspect that the UI won't behave correctly when trying to change the type of a destination from a connectivity statement that I don't own. I didn't test this tho. cc @ddelpiano @Salam-Dalloul

Comment (ddelpiano) [reply frontend/src/components/Forms/StatementForm.tsx:459]: Good catch, I will open a card for @Salam-Dalloul for this. Thanks

Comment (ddelpiano) [reply frontend/src/components/Forms/StatementForm.tsx]: this has been fixed as well

**Review by ddelpiano — APPROVED** [review]

**Pull request merged**